TowersOfHanoi/TowersOfHanoi.cs

```diff
@@ -1,12 +1,270 @@
+        public void MoveBlock(/* int block,  */String towerOrigin, String towerDestination)
+        {
+            // Finds the int value of the originTower top Block object
+            int originBlock = gameboard[towerOrigin].blocks.Peek().block;
+
+            // If the destination tower is empty, pop the top object from the origin tower
+            // and push it onto the the destination stack
+            if (gameboard[towerDestination].blocks.Count == 0)
+            {
+                Block block = gameboard[towerOrigin].blocks.Pop();
+                gameboard[towerDestination].blocks.Push(block);
+                towerMove = "from";
+            }
+            // Runs if the destination tower is not empty
+            else if (gameboard[towerDestination].blocks.Count != 0)
+            {
+                // Finds the int value of the destinationTower top Block object
+                int destinationBlock = gameboard[towerDestination].blocks.Peek().block;
+
+                // Determines if the origin block is smaller than the top block object
+                // of the destination tower 
+                if (originBlock < destinationBlock)
+                {
+                    Block block = gameboard[towerOrigin].blocks.Pop();
+                    gameboard[towerDestination].blocks.Push(block);
+                    towerMove = "from"; // Changes the towerMove back to "from"
+                }
+                else
+                {
+                    Console.WriteLine("Cannot place on a smaller block");
+                }
+            }
+        }
+
+        // Check that tower C is 4 or more blocks
+        public bool CheckForWin()
+        {
+            if (gameboard["C"].blocks.Count == numberOfBlocks)
+            {
+                return true;
+            }
+            return false;
+        }
+    }
+
+    class Tower
+    {
+        public Stack<Block> blocks { get; private set; }
+
+        // Constructor when an int is passed to it. Creates a tower based on the number passed to it
+        public Tower(int numberOfBlocks)
+        {
+            blocks = new Stack<Block>();
+            for (int i = numberOfBlocks; i > 0; i--)
+            {
+                blocks.Push(new Block(i));
+            }
+        }
+
+        // Constructor when nothing is passed. Creates an empty tower
+        public Tower()
+        {
+            blocks = new Stack<Block>();
+        }
+
+        // ToString override to display the the contents of the Block stack in reverse order
+        // so that it displays in the correct order
+        public override String ToString()
+        {
+            String formattedString = "";
+            foreach (Block block in blocks)
+            {
+                formattedString = block.block + " " + formattedString;
+            }
+
+            return formattedString;
+        }
+    }
+
+    class Block
+    {
+        public int block;
```

Comment:
this instance variable being called "block" is a little confusing, maybe "weight" would have been a better choice.